
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Track Types &#8212; HiGlass v1.0 documentation</title>
    <link rel="stylesheet" href="_static/higlass.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     'v1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="View Configs" href="view_config.html" />
    <link rel="prev" title="Server" href="higlass_server.html" />
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-77756807-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-77756807-1');
  </script>
  
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />


  </head>
  <body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="track-types">
<span id="id1"></span><h1>Track Types<a class="headerlink" href="#track-types" title="Permalink to this headline">¶</a></h1>
<div class="section" id="linking-value-scales">
<h2>Linking value scales<a class="headerlink" href="#linking-value-scales" title="Permalink to this headline">¶</a></h2>
<p>The scaling of values in quantitative tracks is done from lowest to highest
values. Two tracks displaying different data will likely have different axis
scales based on the extent of the data visible in each. To enable comparison
between the two tracks, HiGlass supports locking value scales such that they
both display the same axes. This can be done using the track configuration
menu:</p>
<img alt="_images/lock-value-scales.png" class="align-center" src="_images/lock-value-scales.png" />
</div>
<div class="section" id="bed-like">
<h2>Bed-like<a class="headerlink" href="#bed-like" title="Permalink to this headline">¶</a></h2>
<img alt="_images/bedlike-track-thumb.png" class="align-right" src="_images/bedlike-track-thumb.png" />
<p>track-type: <code class="docutils literal"><span class="pre">bedlike</span></code>
datatype: <code class="docutils literal"><span class="pre">bedlike</span></code></p>
<p>The bedlike track is intended to display generic interval data. It is used to
render tracks with a <cite>bedlike</cite> datatype. This usually comes from the <cite>beddb</cite>
filetype. Regular bed-like files can be converted to beddb using the instructions
in the <a class="reference external" href="data_preparation.html#bed-files">data preparation section</a>.</p>
</div>
<div class="section" id="gene-annotations">
<h2>Gene Annotations<a class="headerlink" href="#gene-annotations" title="Permalink to this headline">¶</a></h2>
<img alt="_images/gene-annotations-track-thumb.png" class="align-right" src="_images/gene-annotations-track-thumb.png" />
<p>track-type: <code class="docutils literal"><span class="pre">horizontal-gene-annotations</span></code>
datatype: <code class="docutils literal"><span class="pre">gene-annotations</span></code></p>
<p>Gene annotations display the locations of genes and their exons and introns.
The tracks displayed on HiGlass show a transcript consisting of the union of
all annotated exons in refseq. There are separate tracks for the different
available species. Details on how gene annotation tracks are created is available
in the <a class="reference external" href="gene_annotations.html">gene annotations section</a>.</p>
</div>
<div class="section" id="heatmap">
<h2>Heatmap<a class="headerlink" href="#heatmap" title="Permalink to this headline">¶</a></h2>
<img alt="_images/heatmap-track-thumb.png" class="align-right" src="_images/heatmap-track-thumb.png" />
<p>track-type: <code class="docutils literal"><span class="pre">heatmap</span></code>
datatype: <code class="docutils literal"><span class="pre">matrix</span></code></p>
<p>Heatmaps in HiGlass are usually used to display HiC data. They log-scale input
values and map them to a user-selectable color scale (color map configuration
option). Because HiGlass displays data at varying zoom levels, heatmaps are
displayed at different resolutions depending on the current zoom level. To
limit the resolution of the displayed data, users can set the <cite>Zoom Limit</cite>
configuration option.</p>
</div>
<div class="section" id="d-heatmap">
<h2>1D Heatmap<a class="headerlink" href="#d-heatmap" title="Permalink to this headline">¶</a></h2>
<img alt="_images/horizontal-heatmap-thumb.png" class="align-right" src="_images/horizontal-heatmap-thumb.png" />
<p>track-type: <code class="docutils literal"><span class="pre">horizontal-heatmap</span></code>
datatype: <code class="docutils literal"><span class="pre">matrix</span></code></p>
<p>1D heatmaps are regulat 2D heatmaps sliced across the diagonal and rotated 45
degrees. The base of the 1D heatmap is always the diagonal of the 2D heatmap.
This is useful for displaying data, such as HiC, which has prominent features
along the diagonal.</p>
</div>
<div class="section" id="d-rectangle-domain">
<span id="id2"></span><h2>2D Rectangle Domain<a class="headerlink" href="#d-rectangle-domain" title="Permalink to this headline">¶</a></h2>
<img alt="_images/2d-rectangles-track-thumb.png" class="align-right" src="_images/2d-rectangles-track-thumb.png" />
<p>track-type: <code class="docutils literal"><span class="pre">2d-rectangle-domains</span></code></p>
<p>The rectangle domains track shows rectangular regions on the heatmap. These are
commonly aggregated using clodius based on some importance using the <code class="docutils literal"><span class="pre">clodius</span>
<span class="pre">bedpe</span></code> command. See the <a class="reference external" href="data_preparation.html#bedpe-like-files">data preparation section</a> for an example of the aggregation
command.</p>
<div class="highlight-bash"><div class="highlight"><pre><span></span>higlass-manage ingest --filetype bed2ddb --datatype 2d-rectangle-domains annotations.bed2ddb
</pre></div>
</div>
</div>
<div class="section" id="horizontal-2d-rectangle-domain">
<span id="id4"></span><h2>Horizontal 2D Rectangle Domain<a class="headerlink" href="#horizontal-2d-rectangle-domain" title="Permalink to this headline">¶</a></h2>
<img alt="_images/horizontal-2d-rectangle-domains-thumb.png" class="align-right" src="_images/horizontal-2d-rectangle-domains-thumb.png" />
<p>track-type: <code class="docutils literal"><span class="pre">horizontal-2d-rectangle-domains</span></code>
datatype: <code class="docutils literal"><span class="pre">2d-rectangle-domains</span></code></p>
<p>Horizontal rectangle domains show a 45 degree rotation of rectangular domains
on a 1D track. This track is most commonly used with domains that are close to
the diagonal of the heatmap. Because the track is oriented so that the diagonal
of the 2D view is situated along its base, rectangles far from the diagonal may
be outside of the bounds of the track.</p>
</div>
<div class="section" id="line">
<span id="line-track"></span><h2>Line<a class="headerlink" href="#line" title="Permalink to this headline">¶</a></h2>
<img alt="_images/line-track-thumb.png" class="align-right" src="_images/line-track-thumb.png" />
<p>track-type: <code class="docutils literal"><span class="pre">horizontal-line</span></code>
datatype: <code class="docutils literal"><span class="pre">vector</span></code></p>
<p>Line tracks display 1D vector data. Because each line segment requires two
adjacent points to be present, data with lots of NaNs may lead to a fragmented
plot. For such data, the <a class="reference internal" href="#bar-track"><span class="std std-ref">bar track</span></a> or <a class="reference internal" href="#point-track"><span class="std std-ref">point track</span></a> may be more
appropriate.</p>
</div>
<div class="section" id="bar">
<span id="bar-track"></span><h2>Bar<a class="headerlink" href="#bar" title="Permalink to this headline">¶</a></h2>
<img alt="_images/bar-track-thumb.png" class="align-right" src="_images/bar-track-thumb.png" />
<p>track-type: <code class="docutils literal"><span class="pre">horizontal-bar</span></code>
datatype: <code class="docutils literal"><span class="pre">vector</span></code></p>
<p>Bar tracks display 1D vector data as bars.</p>
</div>
<div class="section" id="point">
<span id="point-track"></span><h2>Point<a class="headerlink" href="#point" title="Permalink to this headline">¶</a></h2>
<img alt="_images/point-track-thumb.png" class="align-right" src="_images/point-track-thumb.png" />
<p>track-type: <code class="docutils literal"><span class="pre">horizontal-point</span></code>
datatype: <code class="docutils literal"><span class="pre">vector</span></code></p>
<p>Point tracks display 1D vector data. Unlike <a class="reference internal" href="#line-track"><span class="std std-ref">line tracks</span></a>,
they are well suited to data with NaNs because they do not require two points
to draw something.</p>
</div>
<div class="section" id="chromosome-labels">
<span id="id5"></span><h2>Chromosome Labels<a class="headerlink" href="#chromosome-labels" title="Permalink to this headline">¶</a></h2>
<img alt="_images/chromosome-labels-thumb.png" class="align-right" src="_images/chromosome-labels-thumb.png" />
<p>track-type: <code class="docutils literal"><span class="pre">horizontal-chromosome-labes</span></code>
datatype: <code class="docutils literal"><span class="pre">chromsizes</span></code>
filetypes: <code class="docutils literal"><span class="pre">chromsizes-tsv</span></code></p>
<p>The chromosome labels track shows the names of the chromosomes. Its data is
sourced from a standard chromSizes file containing chromosome names and
chromosome files. The file can be ingested by the higlass server like any other
tileset. As long as the <cite>datatype</cite> is set to <cite>chromsizes</cite> this track should be
selectable from the “Add Track Dialog”.</p>
</div>
<div class="section" id="chromosome-grid">
<h2>Chromosome Grid<a class="headerlink" href="#chromosome-grid" title="Permalink to this headline">¶</a></h2>
<img alt="_images/chromosome-grid-thumb.png" class="align-right" src="_images/chromosome-grid-thumb.png" />
<p>track-type: <code class="docutils literal"><span class="pre">2d-chromosome-grid</span></code>
datatype: <code class="docutils literal"><span class="pre">chromsizes</span></code>
filetypes: <code class="docutils literal"><span class="pre">chromsizes-tsv</span></code></p>
<p>A chromosome grid displays the boundaries of chromosomes on the 2D area. Its
data is sourced from a standard chromSizes file containing chromosome names and
chromosome files. The file can be ingested by the higlass server like any other
tileset. As long as the <cite>datatype</cite> is set to <cite>chromsizes</cite> this track should be
selectable from the “Add Track Dialog”.</p>
<p>To find the chromosome grid in the list of tracks, search for “chromosomes” when
adding a track to the <em>center</em> view.</p>
</div>
<div class="section" id="stacked-bars">
<h2>Stacked Bars<a class="headerlink" href="#stacked-bars" title="Permalink to this headline">¶</a></h2>
<img alt="_images/horizontal-stacked-bar-scaled-thumb.png" class="align-right" src="_images/horizontal-stacked-bar-scaled-thumb.png" />
<p>track-type: <code class="docutils literal"><span class="pre">horizontal-stacked-bar</span></code>
datatype: <code class="docutils literal"><span class="pre">multivec</span></code></p>
<p>Stacked bar tracks display multivec data. They show multiple values at every
location in the data by using a set of vertically stacked bars. There is an
option to pick ‘unscaled’ and ‘scaled’ representations, which scale the height
of the bars to the maximum and minimum value in all visible tiles or to fit
the height of the track, respectively.</p>
</div>
<div class="section" id="multiple-lines">
<h2>Multiple Lines<a class="headerlink" href="#multiple-lines" title="Permalink to this headline">¶</a></h2>
<img alt="_images/basic-multiple-line-chart-thumb.png" class="align-right" src="_images/basic-multiple-line-chart-thumb.png" />
<p>track-type: <code class="docutils literal"><span class="pre">basic-multiple-line-chart</span></code>
datatype: <code class="docutils literal"><span class="pre">multivec</span></code></p>
<p>Displays multivec data by showing multiple values at every location using a number
of line graphs.</p>
</div>
<div class="section" id="multiple-bar-charts">
<h2>Multiple Bar Charts<a class="headerlink" href="#multiple-bar-charts" title="Permalink to this headline">¶</a></h2>
<img alt="_images/basic-multiple-bar-chart-thumb.png" class="align-right" src="_images/basic-multiple-bar-chart-thumb.png" />
<p>track-type: <code class="docutils literal"><span class="pre">basic-multiple-bar-chart</span></code>
datatype: <code class="docutils literal"><span class="pre">multivec</span></code></p>
<p>Displays multivec data by showing multiple values at every location using a
number of bar graphs.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">HiGlass</a></h1>








<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="atutorial.html">Tutorial</a><ul>
<li class="toctree-l2"><a class="reference internal" href="atutorial.html#prerequisites">Prerequisites</a></li>
<li class="toctree-l2"><a class="reference internal" href="atutorial.html#quickly-viewing-a-dataset">Quickly viewing a dataset</a></li>
<li class="toctree-l2"><a class="reference internal" href="atutorial.html#running-higlass-locally">Running HiGlass locally</a></li>
<li class="toctree-l2"><a class="reference internal" href="atutorial.html#adding-data">Adding data</a></li>
<li class="toctree-l2"><a class="reference internal" href="atutorial.html#viewing-data-in-the-client">Viewing data in the client</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="b_viewer.html">Viewer</a><ul>
<li class="toctree-l2"><a class="reference internal" href="b_viewer.html#adding-views">Adding views</a></li>
<li class="toctree-l2"><a class="reference internal" href="b_viewer.html#closing-and-resizing-views">Closing and resizing views</a></li>
<li class="toctree-l2"><a class="reference internal" href="b_viewer.html#replacing-tracks">Replacing Tracks</a></li>
<li class="toctree-l2"><a class="reference internal" href="b_viewer.html#synchronizing-zoom-and-location">Synchronizing zoom and location</a></li>
<li class="toctree-l2"><a class="reference internal" href="b_viewer.html#adding-cross-sections">Adding cross sections</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="data_preparation.html">Data Preparation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#bed-files">Bed Files</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#bedpe-like-files">Bedpe-like Files</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#bedgraph-files">BedGraph files</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#bigwig-files">bigWig files</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#hitile-files">Hitile files</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#cooler-files">Cooler files</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#multivec-files">Multivec Files</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#epilogos-data-multivec">Epilogos Data (multivec)</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_preparation.html#other-data-multivec">Other Data (multivec)</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="gene_annotations.html">Gene Annotations Tracks</a><ul>
<li class="toctree-l2"><a class="reference internal" href="gene_annotations.html#prerequisites">Prerequisites</a></li>
<li class="toctree-l2"><a class="reference internal" href="gene_annotations.html#creating-the-track">Creating the track</a></li>
<li class="toctree-l2"><a class="reference internal" href="gene_annotations.html#importing-into-higlass">Importing into HiGlass</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="higlass_developer.html">Developer</a><ul>
<li class="toctree-l2"><a class="reference internal" href="higlass_developer.html#embedding-higlass-in-web-page">Embedding HiGlass in web page</a></li>
<li class="toctree-l2"><a class="reference internal" href="higlass_developer.html#public-api">Public API</a></li>
<li class="toctree-l2"><a class="reference internal" href="higlass_developer.html#viewconfs">Viewconfs</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="higlass_docker.html">Docker</a><ul>
<li class="toctree-l2"><a class="reference internal" href="higlass_docker.html#dropping-individual-cache-entries">Dropping individual cache entries</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="higlass_server.html">Server</a><ul>
<li class="toctree-l2"><a class="reference internal" href="higlass_server.html#configuration">Configuration</a></li>
<li class="toctree-l2"><a class="reference internal" href="higlass_server.html#development">Development</a></li>
<li class="toctree-l2"><a class="reference internal" href="higlass_server.html#testing">Testing</a></li>
<li class="toctree-l2"><a class="reference internal" href="higlass_server.html#api">API</a></li>
</ul>
</li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Track Types</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#linking-value-scales">Linking value scales</a></li>
<li class="toctree-l2"><a class="reference internal" href="#bed-like">Bed-like</a></li>
<li class="toctree-l2"><a class="reference internal" href="#gene-annotations">Gene Annotations</a></li>
<li class="toctree-l2"><a class="reference internal" href="#heatmap">Heatmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="#d-heatmap">1D Heatmap</a></li>
<li class="toctree-l2"><a class="reference internal" href="#d-rectangle-domain">2D Rectangle Domain</a></li>
<li class="toctree-l2"><a class="reference internal" href="#horizontal-2d-rectangle-domain">Horizontal 2D Rectangle Domain</a></li>
<li class="toctree-l2"><a class="reference internal" href="#line">Line</a></li>
<li class="toctree-l2"><a class="reference internal" href="#bar">Bar</a></li>
<li class="toctree-l2"><a class="reference internal" href="#point">Point</a></li>
<li class="toctree-l2"><a class="reference internal" href="#chromosome-labels">Chromosome Labels</a></li>
<li class="toctree-l2"><a class="reference internal" href="#chromosome-grid">Chromosome Grid</a></li>
<li class="toctree-l2"><a class="reference internal" href="#stacked-bars">Stacked Bars</a></li>
<li class="toctree-l2"><a class="reference internal" href="#multiple-lines">Multiple Lines</a></li>
<li class="toctree-l2"><a class="reference internal" href="#multiple-bar-charts">Multiple Bar Charts</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="view_config.html">View Configs</a><ul>
<li class="toctree-l2"><a class="reference internal" href="view_config.html#genome-position-search-box">Genome Position Search Box</a></li>
<li class="toctree-l2"><a class="reference internal" href="view_config.html#uids">UIDs</a></li>
<li class="toctree-l2"><a class="reference internal" href="view_config.html#initialxdomain-and-initialydomain">initialXDomain and initialYDomain</a></li>
<li class="toctree-l2"><a class="reference internal" href="view_config.html#tracksourceservers">trackSourceServers</a></li>
<li class="toctree-l2"><a class="reference internal" href="view_config.html#exportviewurl">exportViewUrl</a></li>
<li class="toctree-l2"><a class="reference internal" href="view_config.html#tracks">Tracks</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="views.html">Views</a><ul>
<li class="toctree-l2"><a class="reference internal" href="views.html#genome-position-search-box">Genome Position Search Box</a></li>
<li class="toctree-l2"><a class="reference internal" href="views.html#view-synchronization">View synchronization</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="higlass_server.html" title="previous chapter">Server</a></li>
      <li>Next: <a href="view_config.html" title="next chapter">View Configs</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <div><input type="text" name="q" /></div>
      <div><input type="submit" value="Go" /></div>
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2017,2018 HiGlass Authors.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.6.3</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.10</a>
      
      |
      <a href="_sources/track_types.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>